### PR TITLE
Improved use of Bootstrap

### PR DESCRIPTION
**Pull request by calebsander**
### Summary
Added `flask_bootstrap` to take care of importing Bootstrap. Converted pages to use more Bootstrap and fewer custom styles.

### Test Plan
- Check functionality on the following pages:
  - http://localhost:9000/request
  - http://localhost:9000/login/forgot
  - http://localhost:9000/login
  - http://localhost:9000/marketplace
  - http://localhost:9000/marketplace/search?q=&cat=all
  - http://localhost:9000/marketplace/search?q=&cat=10 (should show no results)
  - http://localhost:9000/marketplace/sell?state=new
  - http://localhost:9000/marketplace/view_item?item_id=9
  - http://localhost:9000/sdfghbvcde5tyjkl (test 404 page)
  - http://localhost:9000/contact

### CODE REVIEW

**Comment by dqu123** [comment]
@calebsander what else does `flask_bootstrap` do besides managing the Bootstrap cdn import?
Also is this ready to merge, or WIP?

**Comment by calebsander** [comment]
`flask_bootstrap` should make it easier to update Bootstrap because it is contained in a `pip` package. I think it also adds a bunch of [`block`s](https://pythonhosted.org/Flask-Bootstrap/basic-usage.html#available-blocks) to the layout that you can modify individually, e.g. setting the title or including a new script. This is done, if you want to merge it.

File: donut/templates/layout.html [hunk 1]
```diff
@@ -1,49 +1,40 @@
...
-        <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css" rel="stylesheet">
-        {# Bootstrap theme #}
+{% block html_attribs %} lang="en"{% endblock %}
+{% block styles %}
+    {{ super() }}
+    <link rel="stylesheet" type="text/css"
+            href="{{ url_for('static', filename='css/style.css') }}" />
+    {# Google Fonts #}
+    <link href='https://fonts.googleapis.com/css?family=EB+Garamond|Roboto' rel='stylesheet' type='text/css'>
+    {# Just for debugging purposes. Don't actually copy this line! #}
```
Comment (dqu123) [inline donut/templates/layout.html]: I guess we don't need this copy pasta

**Review by dqu123 — CHANGES_REQUESTED** [review]

File: donut/templates/layout.html (hunk 1 above)
Comment (calebsander) [reply donut/templates/layout.html]: Yeah that was in there before so I wasn't going to change it

Comment (dqu123) [reply donut/templates/layout.html]: feel free to remove it

**Comment by calebsander** [comment]
Alright, removed the Roboto include since it isn't actually being used and got rid of the IE8 code since IE versions 10 and under represent less than 0.1% of browser usage. Also got rid of the `type="text/css"` attribute since it is set as the `Content-Type` header anyways in the response.

**Review by dqu123 — APPROVED** [review]

**Pull request merged**